secrets/interface.go

```diff
@@ -19,16 +19,23 @@ const (
 type CreateParams struct {
 	ProviderLabel  string
 	Version        int
-	Type           string
+	Type           secrets.SecretType
```

Comment:
I thought we had decided to do opaque secrets only?

Reply:
It's used when creating a secret - you can ask juju to create a "password" that has X chars and symbols etc. The end result is an opaque secret. We can refine this as needed.